psc/src/main/java/com/pinterest/psc/common/event/PscEvent.java

```diff
@@ -0,0 +1,47 @@
+package com.pinterest.psc.common.event;
+
+import com.pinterest.psc.common.TopicUri;
+import com.pinterest.psc.common.TopicUriPartition;
+
+import java.util.Map;
+
+/**
+ * PscEvent framework enables event driven interaction between main loop and psc consumer thread.
+ * At this point it is used only in MemQ consumer case to inform memq consumer that the main loop encountered
+ * the end of a batch while traversing messages from iterator.
+ *
+ * This is an evolving framework.
+ */
+
+public class PscEvent {
+  public final static String EVENT_HEADER = "__EVENT_HEADER";
```

Comment:
I think we need to somehow make sure this internal headers do not overlap. Maybe they should be defined in a central place?

Reply:
Created PscEventHeaders